### ISSUE TITLE
Support for Service Discovery via Consul

### ISSUE THREAD

**Issue by MilgoTest** [LogNet/grpc-spring-boot-starter#119]
Are you going to support service discovery via Consul instead of just Eureka?

**Comment by jvmlet** [comment]
@MilgoTest , there is nothing special in this starter about registering itself in service registry. Its done in configuration file by specifying  the service port to be registered with, that's all. For `consul` I believe you should add its' dependencies and specify `server.port` property as `${grpc.port}` in your configuration file as described in [doc](https://cloud.spring.io/spring-cloud-consul/multi/multi_spring-cloud-consul-discovery.html#_registering_with_consul).

**Comment by MilgoTest** [comment]
But then you are overriding the port for HTTP communication. A better approach is to deliver the grpc port separately. 
And atm you have to do service discovery by yourself (best way would be to support Spring Cloud Gateway or Zuul of course but I guess that's not possible for now) which is a bit clunky because you have to do it everytime for every service which results in a lot of duplicated code.

**Comment by jvmlet** [comment]
@MilgoTest , [here](https://github.com/spring-cloud/spring-cloud-commons/blob/master/docs/src/main/asciidoc/spring-cloud-commons.adoc#serviceregistry) you can find the sample of manually registering the custom service.
If your executable hosts both `http`  and `grpc` services, you can add event listener in your app for `GRpcServerInitializedEvent` event and manually register the running service instance.
In the event handler you have the access  to the running port via passed `getServer()` method, or using `@LocalRunningGrpcPort` annotation on `int` field in your handler. Something like this :
```java
@Component
public class ConsulGrpcRegistration{
  @LocalRunningGrpcPort
   int  runningPort;

    private ServiceRegistry registry;

    public ConsulGrpcRegistration(ServiceRegistry registry) {
        this.registry = registry;
    }

  @Listener
    public void register(GRpcServerInitializedEvent grpcInitEvent) {
    NewService grpcService = new NewService();
   grpcService.setPort(runningPort);    //or grpcService.setPort(grpcInitEvent.getServer().getPort());
   grpcService.setAddress(yourAddress);


     Registration registration = new ConsulRegistration(grpcService );
      this.registry.register(registration);
    }
}
```
Hope  this helps.

**Comment by jvmlet** [comment]
@MilgoTest 
I've just published `3.2.3-SNAPSHOT` with `consul` registration support. 
Can you please configure  snapshot repository 
```
repositories {
    mavenCentral()
   maven { url "https://oss.sonatype.org/content/repositories/snapshots" }

}
dependencies {
    compile 'io.github.lognet:grpc-spring-boot-starter:3.2.3-SNAPSHOT'
}
```

and give it a try ? 
The grpc service will be registered with name `grpc-${spring.application.name}`

**Comment by jvmlet** [comment]
Implemented in `3.3.0`

**Comment by MilgoTest** [comment]
Great update! It's also compatible with Spring Cloud Gateway, nice!
